docs/pbc/all/search/202410.0/base-shop/third-party-integrations/algolia/algolia-personalization.md

```diff
@@ -0,0 +1,151 @@
+and your Algolia search indexes have products. 
+
+## Update Spryker Shop
+
+
+### Install new Spryker packages
+
+```bash
+composer install spryker-shop/traceable-event-widget
+```
+
+### Update Spryker packages
+
+```bash
+composer update --with-dependencies spryker/search-http spryker/customer \
+spryker-shop/shop-ui spryker-shop/catalog-page spryker-shop/cart-page spryker-shop/checkout-page spryker-shop/home-page \
+spryker-shop/product-detail-page spryker-shop/product-group-widget spryker-shop/product-set-detail-page spryker-shop/quick-order-page
+```
+
+if the command does now work, try it with `--with-all-dependencies` flag instead. 
+
+### Enabled new features
+1. Update project config:
+```php
+// config_default.php
+
+$config[KernelAppConstants::TENANT_IDENTIFIER]
+    //..   
+    = getenv('SPRYKER_TENANT_IDENTIFIER') ?: '';
+```
+
+2. Enable new widget, that will trigger events for user actions and send them to [Algolia Insights](https://www.algolia.com/doc/guides/sending-events/getting-started/): 
+```php
+// src/Pyz/Yves/ShopApplication/ShopApplicationDependencyProvider.php
+
+class ShopApplicationDependencyProvider extends SprykerShopApplicationDependencyProvider
+{
+
+    protected function getGlobalWidgets(): array
+    {
+        return [
+            //...
+            SprykerShop\Yves\TraceableEventWidget\Widget\TraceableEventWidget::class
+        ];
+}
+```
+
+3. Enabled the plugin that will generate an anonymous token for guest users in the session:
+```php
+// src/Pyz/Yves/EventDispatcher/EventDispatcherDependencyProvider.php
+
+class EventDispatcherDependencyProvider extends SprykerEventDispatcherDependencyProvider
+{
+
+    protected function getEventDispatcherPlugins(): array
+    {
+        return [
+            //...
+            new Spryker\Yves\Customer\Plugin\EventDispatcher\AnonymousIdSessionAssignEventDispatcherPlugin(),
+        ];
+}
+```
+
+4. Build a new javascript assets for Yves `npm ci && npm run yves` or `console frontend:project:install-dependencies && console frontend:yves:build`.
+   Usually it's executed automatically during Spryker Cloud deployment pipeline. But it's better to check this command on local development environment first.
+
+5. Check your Yves's compatibility with the feature:
+
+In case of customizations your codebase could have updated Yves templates on project level (src/Pyz/Yves/).
+It could be a reason that some events won't be triggered or triggered with incorrect data.
+
+// TODO (@matweew/@supproduction): updated after the implementation of debugger.
+`TraceableEventWidgetConfig::getIsDebugEnabled()` set to `true`.
+  
+* Run the project locally or deploy to testing environment.
+* Open your Shop Storefront home page.
+* Open browser's development console and check "Preserve log" in the settings.
+
+Test the correctness of data in the triggered events in the browser console:
+* Open Home page
+  * (if applicable) Click on a product - `PRODUCT_CLICK`
```

Comment:
When are those events applicable?

Reply:
I mean if applicable for the project's home page (if they products are on the home page). Not clear?